packages/core/src/__test__/integration/two-controlled-props.test.ts

```diff
@@ -0,0 +1,129 @@
+        ({ pageIndex }: State) => ({ pageIndex }),
+        pageIndexVM,
+      );
+      store.subscribe((state) => pageCountSelector(state));
+      store.subscribe((state) => pageIndexSelector(state));
+      return {
+        store,
+        pageCountVM,
+        pageIndexVM,
+      };
+    }
+    it('pageCount and pageIndex controlled', () => {
+      const { store, pageCountVM, pageIndexVM } = createComponent({
+        pageCount: {
+          controlledMode: true,
+          changeCallback: (newPageCount: number) => {
+            store.addUpdate((state) => ({
+              ...state,
+              pageCount: newPageCount + 2,
+            }));
+            store.commitPropsUpdates();
+          },
+        },
+        pageIndex: {
+          controlledMode: true,
+          changeCallback: (newIndex: number) => {
+            store.addUpdate((state) => ({
+              ...state,
+              pageIndex: newIndex + 1,
+            }));
+            store.commitPropsUpdates();
+          },
+        },
+      });
+      expect(store.getState()).toMatchObject(initialState);
+
+      store.addUpdate((state) => ({
+        ...state,
+        pageCount: 5,
+        pageIndex: 5,
+      }));
+      store.commitUpdates();
+
+      expect(store.getState()).toMatchObject({ pageCount: 7, pageIndex: 6 });
+      expect(pageCountVM).toBeCalledTimes(1);
+      expect(pageCountVM).toBeCalledWith({ pageCount: 7 });
+      expect(pageIndexVM).toBeCalledTimes(2);
+      expect(pageIndexVM).toBeCalledWith({ pageIndex: 6 });
+    });
+    it('pageCount controlled pageIndex uncontrolled', () => {
+      const { store, pageCountVM, pageIndexVM } = createComponent({
+        pageCount: {
+          controlledMode: true,
+          changeCallback: (newPageCount: number) => {
+            store.addUpdate((state) => ({
+              ...state,
+              pageCount: newPageCount - 2,
+            }));
+            // simulate component logic: pageIndex should be less or equal ageCount
+            store.addUpdate((state) => ({
+              ...state,
+              pageIndex: newPageCount - 2,
+            }));
+            store.commitPropsUpdates();
+          },
+        },
+      });
+      expect(store.getState()).toMatchObject(initialState);
+      store.addUpdate((state) => ({
+        ...state,
+        pageCount: 5,
+        pageIndex: 5,
+      }));
+      store.commitUpdates();
+
+      expect(store.getState()).toMatchObject({ pageCount: 3, pageIndex: 3 });
+      expect(pageCountVM).toBeCalledTimes(2);
+      expect(pageIndexVM).toHaveBeenNthCalledWith(1, { pageIndex: 5 });
+      expect(pageCountVM).toHaveBeenNthCalledWith(2, { pageCount: 3 });
+      expect(pageIndexVM).toBeCalledTimes(2);
+      expect(pageIndexVM).toHaveBeenNthCalledWith(1, { pageIndex: 5 });
+      expect(pageIndexVM).toHaveBeenNthCalledWith(2, { pageIndex: 3 });
+    });
+    it.skip('pageCount controlled pageIndex uncontrolled, wrong PageIndex after change', () => {
```

Comment:
Let's add an empty line between test cases.

```suggestion
    });
    
    it.skip('pageCount controlled pageIndex uncontrolled, wrong PageIndex after change', () => {
```